pkg/build/graph/edges.go

```diff
@@ -55,24 +57,37 @@ func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildCon
 		g.AddEdge(in, node, BuildInputEdgeKind)
 	}
 
+	sourceImages := []*kapi.ObjectReference{}
 	from := buildutil.GetImageStreamForStrategy(node.BuildConfig.Spec.Strategy)
 	if from != nil {
-		switch from.Kind {
+		sourceImages = append(sourceImages, from)
```

Comment:
Hmm, not sure how I missed this difference. I prefer having a single behavior for build-chain so lets make this correct.